contracts/strategy/liquity/LiquityStrategy.sol

```diff
@@ -0,0 +1,319 @@
+        if (_newAdmin == address(0x0)) revert StrategyAdminCannotBe0Address();
+
+        if (_newAdmin == msg.sender)
+            revert StrategyCannotTransferAdminRightsToSelf();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _newAdmin);
+        _setupRole(MANAGER_ROLE, _newAdmin);
+
+        _revokeRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _revokeRole(MANAGER_ROLE, msg.sender);
+    }
+
+    //
+    // IStrategy
+    //
+
+    /**
+     * Returns true since strategy is synchronous.
+     */
+    function isSync() external pure override(IStrategy) returns (bool) {
+        return true;
+    }
+
+    /// @inheritdoc IStrategy
+    function hasAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        return investedAssets() > 0;
+    }
+
+    /// @inheritdoc IStrategy
+    /// @notice ETH & LQTY rewards of the strategy waiting to be claimed in the liquity stability pool are not included
+    function investedAssets()
+        public
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        return stabilityPool.getCompoundedLUSDDeposit(address(this));
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        stabilityPool.provideToSP(balance, address(0));
+
+        emit StrategyInvested(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    /// @notice will also claim unclaimed LQTY & ETH gains
+    /// @notice when amount > total deposited, all available funds will be withdrawn
+    function withdrawToVault(uint256 amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (amount == 0) revert StrategyAmountZero();
+        if (amount > investedAssets()) revert StrategyNotEnoughShares();
+
+        // withdraws underlying amount and claims LQTY & ETH rewards
+        stabilityPool.withdrawFromSP(amount);
+
+        uint256 lqtyRewards = lqty.balanceOf(address(this));
+        uint256 ethRewards = address(this).balance;
+        // backend to keep track of this and decide when to call reinvestRewards
+        emit StrategyRewardsClaimed(lqtyRewards, ethRewards);
+
+        // use balance instead of amount since amount could be greater than what was actually withdrawn
+        uint256 balance = underlying.balanceOf(address(this));
+        if (!underlying.transfer(vault, balance)) {
+            revert TokenTransferFailed(address(underlying));
+        }
+
+        emit StrategyWithdrawn(balance);
+    }
+
+    /**
+     * Collects the LQTY & ETH rewards from the stability pool, swaps the rewards to LUSD,
+     * and reinvests swapped LUSD amount into the stability pool to create compound interest on future gains.
+     *
+     * @notice Can only be called by the account with the ADMIN role.
```

Comment:
I don't see a `onlyAdmin` anywhere on this function

Reply:
the idea was to save some gas by not having the `onlyAdmin` modifier on both `harvest` and `reinvestRewards` because `harvest` implicitly calls the `reinvestRewards` function. The transaction will fail in any case if the caller is not admin

Reply:
Hmm I cant think of any security vulnerabilities if the harvest & reInvestRewards can be called by anyone.

Reply:
Currently, only the admin can claim the rewards because if anyone else calls this the transaction will fail.

Reply:
Oh, yeah, because the other function is always executed. I don't like not seeing the safeguard in every public function, but I understand the reasoning; as long as there are tests, let's roll with it.